### ISSUE TITLE
Scivolemo pri Ĝue

### ISSUE THREAD

**Issue by ortegacmanuel** [psychoslave/gxue#1]
@psychoslave saluton! kion vi celas per Gxue? mi ĝojas ekscii ke vi programas per Ruby, mi amas ĝin kaj mi ŝatus labori pri la tradukado de la rubenaj ŝlosilvortoj al Esperanto..mi jam faras kelkajn provojn:

```
se 4 > 5
  eligu 'la kondiĉo veras'
male
  eligu 'la kondiĉo malveras'
fino
```

**Comment by psychoslave** [comment]
Saluton @ortegacmanuel 

Interesan provon!

Per Ĝue (Github ne permesas uzi ne-ASKIan nomon) mi provas plibonigi mian konon de Ruby kaj samtempe provi ideojn ke mi havas pri tute Esperanta programlingvo, opcie tute laŭlitera, sed ĉiam celi la pli kurtaj eldiroj ke esperanto permesas.

Mi certe ŝatus ke oni povas lokaĵigi Ruby tiel ke vi proponas, sed mi kredas ke la nuna funkciigo de Ruby ne facile permisos tiom da traduko. La ĉefleksoj (*keyword*) ne povas esti alinomita sen ŝanĝi la disponigilo (*parser*) de Ruby. Ĉu vi havus lertecon por tio? Eble mi mem zorgos pri tiu se mi trovos la tempon kaj lertecon, sed antaŭ mi komencas per la pli facile funkcigeblaj trajtoj.

**Comment by psychoslave** [comment]
Ankaŭ, tie estas alia propono
```
se 4 > 5
  eligu 'la kondiĉo veras'
alie
  eligu 'la kondiĉo malveras'
ope
```

Fakte mi pensis pri *ope* ĉar ĝi temas pri grupi, tie grupi la ordonojn. Sed ankaŭ estas uzi la interjekcio *hop*, ĉar ĝi temas pri eta salto, tie salto de konteksto.

Kion opinias vi?

**Comment by psychoslave** [comment]
Mi aldonis pli precizajn celojn en la priskribo de la projekto.

**Comment by ortegacmanuel** [comment]
@psychoslave dankon pro via respondo, _ope_ kaj _hop_ ne plene konvinkas min sed ja la fakto estas ke `fino` tre tre malplaĉas al mi, nu, la serĉado de alternativo estas prioritato por mi...mi daŭre pripensos pri ope kaj hop...hop ja pli plaĉas al mi ...

Mi havas la lertecon sed mankas al mi la tempo...tuj mi havos iom da tempo mi laboros pri tio..esporu [tiun ĉi projekton](https://github.com/jjuliano/bato)

**Comment by psychoslave** [comment]
Oh, mirinda, la projekto ke vi ligis ŝajnas tre bona, avangarda kaj progresema!

Unu avantaĝo de *hop* estas ke ĝi estas facile ĉenvoĉebla: "hop hop hop". :laughing: Kiel oni ofte fermas kuntekstoj en Ruby per "end", tiu estas interesa se oni deziras dikti kodon laŭvoĉe.

**Comment by psychoslave** [comment]
Nu, mi startis legi la kodon de bato. Ŝajnas ke la pli interesaj partoj estas en
- https://github.com/jjuliano/bato/blob/master/lib/bato/ruby_parser_patches.rb
- https://github.com/jjuliano/bato/blob/master/lib/bato/tagatala.rb

Resume ĝi uzas 'ruby2ruby' kaj 'ruby_parser' per simbol-esprimoj (*S-expression*).

Aliaj interesaj ligoj:
- https://ruby-doc.org/stdlib-2.0.0/libdoc/ripper/rdoc/Ripper.html
- https://github.com/seattlerb/ruby_parser/blob/588c7650ce4e111800c1d4c9f3b86f3bc11ebb67/lib/ruby_parser_extras.rb

Do, mi ne jam tute komprenis kiel ĝi funkcias, sed ŝajnas ke ĝi ne bezonas multe da kodo.

Persone, mi deziras ke Ĝue povas esti uzita ambaŭ kiel REPL-ilo kaj por ruli dosierojn. Kion opinias vi? Estos necesa difini dosier-finaĵo, eble simple `.ĝue`.

**Comment by psychoslave** [comment]
@ortegacmanuel bonvolu diri al mi kion vi opinias pri la celoj ke mi skribis en  https://github.com/psychoslave/gxue/blob/master/PRISKRIBO.md. :pray:
